src/routes/join/components/SignupPage.js

```diff
@@ -0,0 +1,180 @@
+const React = require('react');
+const T = require('prop-types');
+const { default: Styled } = require('styled-components');
+const { default: Typography } = require('@material-ui/core/Typography');
+const { default: TextField } = require('@material-ui/core/TextField');
+const { default: Button } = require('@material-ui/core/Button');
+const { default: Box } = require('@material-ui/core/Box');
+const { default: Link } = require('@material-ui/core/Link');
+const { default: FormControl } = require('@material-ui/core/FormControl');
+const { default: InputLabel } = require('@material-ui/core/InputLabel');
+const { default: Input } = require('@material-ui/core/Input');
+const { default: InputAdornment } = require('@material-ui/core/InputAdornment');
+const { default: IconButton } = require('@material-ui/core/IconButton');
+const { default: Visibility } = require('@material-ui/icons/Visibility');
+const { default: VisibilityOff } = require('@material-ui/icons/VisibilityOff');
+const StrangeForms = require('strange-forms');
+
+const internals = {};
+
+module.exports = class SignupPage extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        reqCreateAccount: T.func.isRequired
+    };
+
+    static fields = {
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: ''
+    };
+
+    defaultSubmit = (ev) => {
```

Comment:
I personally find it nice to keep methods below class constructors, not sure what others think!

Reply:
Actually, looks like this isn't used anyway :)